### DISCUSSION TITLE
Adding macro property to custom python script

### DISCUSSION THREAD

**Discussion by DVark09** [WarmUpTill/SceneSwitcher#1256] [Q&A] [answered]
Perhaps I have glanced over something in the scripting wiki page, but how do I add macro properties to conditions/actions via the current API? 
I'm trying to make a script that gives me the process' executable name and window title (the latter of which is possible to get already) so that I can set a window capture to the currently active window.
I tried using set_variable_value in my condition function but it just fails. I checked all the available example codes and I don't see any of them creating a macro property.
Is there a way to do this? Doesn't matter if it's getting it via macro property or condition/action value.
![image](https://github.com/user-attachments/assets/f2ff1686-c70d-4c96-889e-252c403eccb0)

**Comment by DVark09** [comment]
I wasn't able to run a python/powershell script by itself to fix this since those spawn in a window in itself and those get picked up instead. I made a workaround that gets all 3 attributes and puts them in one env var that I can feed directly to my window capture.

**Comment by WarmUpTill** [comment, marked as answer]
> how do I add macro properties to conditions/actions via the current API?

Unfortunately, currently the scripting API does not expose a way to register macro properties.
It only allows you to directly modify variables you have defined.

So you can do something like this from within a script:
![image](https://github.com/user-attachments/assets/2606e5ca-2c5b-42e8-9269-bbe6f8c5031e)

I will try to add the option to register macro properties as well when I find the time.

**Reply by DVark09** [reply]
Thank you, It would be nice if we had at this since running powershell/python scripts spawns their own window which would be an issue on this one exact scenario at least.

**Reply by WarmUpTill** [reply]
I got around to implementing the option to register "macro properties" via the scripting API.

![image](https://github.com/user-attachments/assets/16b2da1f-9558-4848-83f9-f3e89ed4f4e2)

A build will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/11862171910

You can find examples how to use the new API here:
https://github.com/WarmUpTill/SceneSwitcher/blob/script-temp-var/scripting/examples.py#L94-L111
https://github.com/WarmUpTill/SceneSwitcher/blob/script-temp-var/scripting/examples.lua#L95-L111

Let me know if you have any questions or doubts about the implementation.

**Reply by DVark09** [reply]
I tested it and it works perfectly for me. Thanks for letting me know!
